library/core/src/net/socket_addr.rs

```diff
@@ -42,10 +39,6 @@ pub enum SocketAddr {
 ///
 /// See [`SocketAddr`] for a type encompassing both IPv4 and IPv6 socket addresses.
 ///
-/// The size of a `SocketAddrV4` struct may vary depending on the target operating
-/// system. Do not assume that this type has the same memory layout as the underlying
-/// system representation.
```

Comment:
I'm ok with removing the size comment, but isn't the other layout note still relevant? (e.g. don't try to treat this like a C `struct sockaddr`!)

Reply:
So, it's technically relevant, but it feels unprecedented in the standard library, since no other type adds a comment like this. I actually took a bit to look through the existing types, though, and I guess these are the types where this would technically be most applicable. (You wouldn't expect a `File` to be equivalent to a file handler, for example.)

Like, most people are not going to try transmuting random standard library structs to libc structs unless it's explicitly marked as allowed, and since transmuting is unsafe, they're also kind of admitting fault if they do this.

But also, is there actually a reason we *couldn't* make this equivalent to a C `sockaddr` anyway? Since no padding is required, and we don't really gain anything from letting the fields be randomly arranged.

Reply:
We used to wrap `sockaddr` directly, and prominent crates did make unsafe assumptions about that. We changed to Rust-native types in #78802, which also documented many of those problems along the way.

So yes, in general users shouldn't make any assumptions about the underlying type, but they **did** in this case. Therefore, I think it's worth keeping some warning in the docs, even though it would be obviously broken if you tried it anymore.

Reply:
So, for some reason, I had naïvely assumed that `sockaddr` just used the IETF definition on all platforms, which is clearly not correct. And besides that, it's not the same definition as the one we use.

I'll update to remark on this.